### PR TITLE
fix: panic when setting config value

### PR DESCRIPTION
**Pull request by roele**
Rough draft around the discussion #3820. 

This handles the basic use-case and will behave like the following.
```
# set simple value or nested value
mise config set tools.node 20 --> node = "20"
mise config set tools.node.version 20 --> node = { version = "20" }

# transform version to inline table if its already set
mise config set tools.node 20 --> node = "20"
mise config set tools.node.postinstall 'corepack enable' --> node = { version = "20", postinstall = "corepack enable" }
```

### CODE REVIEW

File: src/cli/config/set.rs [hunk 1]
```diff
@@ -99,7 +112,13 @@ impl ConfigSet {
                 TomlValueTypes::Infer => bail!("Type not found"),
             };
 
-            container.as_table_mut().unwrap().insert(last_key, value);
+            container
+                .as_table_like_mut()
+                .unwrap_or({
+                    let t = toml_edit::Table::new();
+                    toml_edit::Item::Table(t).as_table_like_mut().unwrap()
```
Comment (jdx) [inline src/cli/config/set.rs:123]: I think you have this in another PR but we should set the flag for implicit keys or whatever it's called

**Comment by jdx** [comment]
mind adding an e2e test? it doesn't look like we have any for these commands, but I think it should be pretty simple. I would copy https://github.com/jdx/mise/blob/main/e2e/cli/test_settings_add

**Comment by roele** [comment]
will add a test and look into the additional type inference issue but this might have to wait until tomorrow

**Comment by jdx** [comment]
sounds good, I think what you have in the description is good enough. A pattern I've been doing is this which might be useful here:

https://github.com/jdx/mise/blob/b66f7cac26186b1f935c11e296647baebf1a20f1/e2e/lockfile/test_lockfile_use#L14-L16

it's basically a snapshot test

File: src/cli/config/set.rs [hunk 2]
```diff
@@ -78,7 +91,10 @@ impl ConfigSet {
                             SettingsType::ListString => TomlValueTypes::List,
                             SettingsType::ListPath => TomlValueTypes::List,
                         },
-                        None => TomlValueTypes::String,
+                        None => match self.value.as_str() {
+                            "true" | "false" => TomlValueTypes::Bool,
```
Comment (roele) [inline src/cli/config/set.rs:95]: not sure if we also should treat `0,1,y,n` as `boolean`, would these even map to TOML values? mid-term the whole type inference logic should probably be moved to a package as it is also used by `settings set` for example

**Pull request merged**